crates/matrix-sdk/src/client/futures.rs

```diff
@@ -77,6 +77,13 @@ impl<R> SendRequest<R> {
         self
     }
 
+    /// Use the given [`RequestConfig`] for this send request, instead of the
+    /// one provided by default.
+    pub fn with_request_config(mut self, request_config: Option<RequestConfig>) -> Self {
```

Comment:
Why would we take an option here? Not calling this is the same as passing `None`, no?

Reply:
It's to avoid the messy pattern described in OP and commit message. I was a bit torn about this.

Reply:
Could do `impl Into<Option<RequestConfig>>` so people can pass a `RequestConfig` directly. There's no reason to ever call it with `None` so type inference should not be an issue.

Reply:
That's a good idea.